packages/uniforms-bootstrap4/__tests__/SelectField.tsx

```diff
@@ -325,6 +325,117 @@ test('<SelectField> - disabled items (options) based on predicate', () => {
+  const wrapper = mount(
+    element,
+    createContext({
+      x: { type: Array, allowedValues: ['a', 'b'] },
+      'x.$': { type: String },
+    }),
+  );
+
+  expect(wrapper.find('select')).toHaveLength(1);
+  expect(wrapper.find('select').prop('value')).toStrictEqual([]);
+});
+
+test('<SelectField> - renders a select with correct value (model)', () => {
+  const element = <SelectField name="x" />;
+  const wrapper = mount(
+    element,
+    createContext(
+      {
+        x: { type: Array, allowedValues: ['a', 'b'] },
+        'x.$': { type: String },
+      },
+      { model: { x: ['b'] } },
+    ),
+  );
+
+  expect(wrapper.find('select')).toHaveLength(1);
+  expect(wrapper.find('select').prop('value')).toStrictEqual(['b']);
+});
+
+test('<SelectField> - renders a select with correct value (specified)', () => {
+  const element = <SelectField name="x" value={['b']} />;
+  const wrapper = mount(
+    element,
+    createContext({
+      x: { type: Array, allowedValues: ['a', 'b'] },
+      'x.$': { type: String },
+    }),
+  );
+
+  expect(wrapper.find('select')).toHaveLength(1);
+  expect(wrapper.find('select').prop('value')).toStrictEqual(['b']);
+});
+
+test('<SelectField> - renders a select which correctly reacts on change (first value)', () => {
+  const onChange = jest.fn();
+
+  const element = <SelectField name="x" />;
+  const wrapper = mount(
+    element,
+    createContext(
+      {
+        x: { type: Array, allowedValues: ['a', 'b'] },
+        'x.$': { type: String },
+      },
+      { onChange },
+    ),
+  );
+
+  expect(wrapper.find('select')).toHaveLength(1);
+  expect(
+    wrapper.find('select').simulate('change', { target: { value: 'a' } }),
+  ).toBeTruthy();
+  expect(onChange).toHaveBeenLastCalledWith('x', ['a']);
+});
+
+test('<SelectField> - renders a select which correctly reacts on change (next value)', () => {
+  const onChange = jest.fn();
+
+  const element = <SelectField name="x" value={['b']} />;
+  const wrapper = mount(
+    element,
+    createContext(
+      {
+        x: { type: Array, allowedValues: ['a', 'b'] },
+        'x.$': { type: String },
+      },
+      { onChange },
+    ),
+  );
+
+  expect(wrapper.find('select')).toHaveLength(1);
+  expect(
+    wrapper.find('select').simulate('change', { target: { value: 'a' } }),
+  ).toBeTruthy();
+  expect(onChange).toHaveBeenLastCalledWith('x', ['a', 'b']);
+});
+
+test('<SelectField> - renders a select which correctly reacts on change (uncheck)', () => {
+  const onChange = jest.fn();
+
+  const element = <SelectField name="x" value={['a']} />;
+  const wrapper = mount(
+    element,
+    createContext(
+      {
+        x: { type: Array, allowedValues: ['a', 'b'] },
+        'x.$': { type: String },
+      },
+      { onChange },
+    ),
+  );
+
+  expect(wrapper.find('select')).toHaveLength(1);
+  expect(
+    wrapper.find('select').simulate('change', { target: { value: 'a' } }),
+  ).toBeTruthy();
```

Comment:
Is this test correct? It doesn't need to include `selectedIndex`?

Reply:
Seems like uncheck the last checked element generates a target without value. Indeed we have to add a target with `selectedIndex` to represent real behavior.